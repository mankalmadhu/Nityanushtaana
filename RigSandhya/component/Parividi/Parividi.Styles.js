import {
 StyleSheet
} from 'react-native';


const parividiVinyasa = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
  separator: {
    flex: 1,
    height: StyleSheet.hairlineWidth,
    backgroundColor: '#8E8E8E',
  },
});

export default parividiVinyasa;