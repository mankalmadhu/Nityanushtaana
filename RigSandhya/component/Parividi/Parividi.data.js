/////////////////////////////////////
const prakaranaMahiti = [
  {
    "title":  "ಪ್ರಾತಸ್ಸ೦ಧ್ಯಾ",
    "picture": "./chitra/sandhya.png"
  },
  {
    "title":  "ಪ್ರಾತಸ್ಸಮಿಧಾದಾನ",
    "picture": "./chitra/agnikarya.png"
  },
  {
    "title":  "ಮಾಧ್ಯಾಹ್ನಿಕಸ೦ಧ್ಯಾ",
    "picture": "./chitra/sandhya.png"
  },
  {
    "title":  "ಸಾಯ೦ಸ೦ಧ್ಯಾ",
    "picture": "./chitra/sandhya.png"
  },
  {
    "title":  "ಸಾಯ೦ಸಮಿಧಾದಾನ",
    "picture": "./chitra/agnikarya.png"
  },
  {
    "title":  "ಯಜ್ಞೋಪವೀತ",
    "picture": "./chitra/upadesha.png"
  },
  {
    "title":  "ಉಪನಯನಕಾರಿಕ",
    "picture": "./chitra/upadesha.png"
  }
  ]
////////////////////////////////////////
export default prakaranaMahiti;