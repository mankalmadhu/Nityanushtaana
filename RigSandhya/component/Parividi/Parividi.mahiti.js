
export const sandhyaChitra = () => require('./chitra/sandhya.png');
export const agnikaryaChitra = () => require('./chitra/agnikarya.png');
export const upadeshaChitra = () => require('./chitra/upadesha.png');

export const parividiMahiti = () => [
  {
    "shershike":  "ಪ್ರಾತಸ್ಸ೦ಧ್ಯಾ",
    "chitra": exports.sandhyaChitra()
  },
  {
    "shershike":  "ಪ್ರಾತಸ್ಸಮಿಧಾದಾನ",
    "chitra": exports.agnikaryaChitra()
  },
  {
    "shershike":  "ಮಾಧ್ಯಾಹ್ನಿಕಸ೦ಧ್ಯಾ",
    "chitra": exports.sandhyaChitra()
  },
  {
    "shershike":  "ಸಾಯ೦ಸ೦ಧ್ಯಾ",
    "chitra": exports.sandhyaChitra()
  },
  {
    "shershike":  "ಸಾಯ೦ಸಮಿಧಾದಾನ",
    "chitra": exports.agnikaryaChitra()
  },
  {
    "shershike":  "ಯಜ್ಞೋಪವೀತ",
    "chitra": exports.upadeshaChitra()
  },
  {
    "shershike":  "ಉಪನಯನಕಾರಿಕ",
    "chitra": exports.upadeshaChitra()
  }
  ]
