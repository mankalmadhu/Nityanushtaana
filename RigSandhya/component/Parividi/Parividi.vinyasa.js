import {StyleSheet} from 'react-native';

export default parividiVinyasa = StyleSheet.create({
  avarana: {
    flex: 3,
    backgroundColor: '#333333'
  },
  vibhajaka: {
    flex: 1,
    height: StyleSheet.hairlineWidth,
   },
});