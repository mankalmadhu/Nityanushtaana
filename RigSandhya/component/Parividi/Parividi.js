import {ListView, View} from 'react-native';
import React from 'react';

import Adhyaya from '../Adhyaya/Adhyaya';
import ParividiShershike from '../ParividiShershike/ParividiShershike';
import {parividiVinyasa} from './Parividi.vinyasa';
import {parividiMahiti} from './Parividi.mahiti';

export default class Parividi extends React.Component {
  constructor(props) {
    super(props);

    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this.state = {
      dataSource: ds.cloneWithRows(parividiMahiti),
    };
  }
  render() {
    return (
      <ListView
        style={parividiVinyasa.avarana}
        dataSource={this.state.dataSource}
        renderHeader={() => <ParividiShershike />}
        renderRow={(parividiMahiti) => <Adhyaya {...parividiMahiti} />}
        renderSeparator={(sectionId,rowId) => <View key={rowId} style={parividiVinyasa.vibhajaka} />}
        
      />
    );
  }
}