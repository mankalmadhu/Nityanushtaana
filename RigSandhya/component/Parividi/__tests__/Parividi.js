import 'react-native';
import React from 'react';
import Parividi from '../Parividi';

import Renderer from 'react-test-renderer';

jest.mock('../../Adhyaya/Adhyaya');
jest.mock('../../ParividiShershike/ParividiShershike');
jest.mock('../Parividi.mahiti',()=>{
  return {
    
    parividiMahiti: [{ 
        "shershike":  "MockTitle1",
        "chitra": 'SomePicture1'
      },
      { 
        "shershike":  "MockTitle2",
        "chitra": 'SomePicture2'
      }]
}});

test("Parividi" , () => {
   const parividi = Renderer.create(
       <Parividi/>
   ).toJSON();
   expect(parividi).toMatchSnapshot();
});