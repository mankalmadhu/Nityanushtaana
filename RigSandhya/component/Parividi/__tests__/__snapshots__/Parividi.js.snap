// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parividi 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  renderSeparator={[Function]}
  style={
    Object {
      "backgroundColor": "#333333",
      "flex": 3,
    }
  }
>
  <View>
    <div>
      Adhyaya 
      MockTitle1 SomePicture1
    </div>
    <div>
      Adhyaya 
      MockTitle2 SomePicture2
    </div>
  </View>
</RCTScrollView>
`;
